sdk/servicebus/azure-servicebus/azure/servicebus/aio/_async_auto_lock_renewer.py

```diff
@@ -147,9 +162,23 @@ def register(
         if self._shutdown.is_set():
             raise ServiceBusError("The AutoLockRenewer has already been shutdown. Please create a new instance for"
                                   " auto lock renewing.")
-        starttime = renewable_start_time(renewable)
+
+        starttime = get_renewable_start_time(renewable)
+
+        # This is a heuristic to compensate if it appears the user has a lock duration less than our base renew period
+        time_until_expiry = get_renewable_lock_duration(renewable)
+        renew_period_override = None
+        # Default is 10 seconds, but let's leave ourselves a small margin of error because clock skew is a real problem
+        if time_until_expiry <= datetime.timedelta(seconds=self._renew_period + 1):
```

Comment:
I see in the sync code the margin is 0.5 while here it's 1, I guess it's an empirical value from your experients?

Reply:
Ah, it was initially 1, I lowered it to .5 while writing unit tests, should normalize the two (into a constant likely to avoid this), truthfully any non-zero non-trivial value (e.g. 1-2s) is likely sufficient.  For obvious reasons determining exact margin of clock skew is "an interesting problem" so I've only hand-waved at order-of-magnitude thus far, and it can likely be improved iteratively.